symengine/test_visitors.h

```diff
@@ -24,14 +29,17 @@ class ZeroVisitor : public BaseVisitor<ZeroVisitor>
     void bvisit(const Set &x)
     {
         is_zero_ = tribool::trifalse;
+        neither_ = true;
```

Comment:
I'm wondering whether these should raise an Exception instead of silently returning false

Reply:
I have been thinking and I think you are right. If all tests that work with numbers should throw for all non-number objects this would both make the tests slightly simpler, faster and also safer for the user. I will have to fix this for all test_visitors so I might merge this and then revisit all visitors in a new PR.